packets/packets.go

```diff
@@ -39,6 +39,7 @@ type (
 		Unpack(*bytes.Buffer) error
 		Buffers() net.Buffers
 		WriteTo(io.Writer) (int64, error)
+		ToControlPacket() *ControlPacket
```

Comment:
Adding this couples the interface to this specific implementation (prior to this change it was generic, only utilising the standard library). While its not currently a goal, I suspect that at some point in the future we may want to add a very basic v3 implementation and this change would complicate that (because it would limit the ability to implement `Packet` in a different package).
I believe that this is probably why the functions were implemented as they were (a bit of duplication keeps things loosely coupled).
We can make this change but do need to consider the potential future impact.